### PR TITLE
Equal in unit tests

### PR DESCRIPTION
**Pull request by jtlap**
whenever floating are used and result can differ due to IEEE computation  STF_EQUAL(...) is replaced by STF_ULP_EQUAL(...,0.5)

### CODE REVIEW

File: test/function/scalar/abs.cpp [hunk 1]
```diff
@@ -61,13 +61,13 @@ STF_CASE_TPL( "Check std abs behavior with floating", STF_IEEE_TYPES )
   STF_TYPE_IS(r_t, T);
 
 #ifndef BOOST_SIMD_NO_INVALIDS
-  STF_EQUAL(bs::std_(abs)(bs::Inf<T>()),  bs::Inf<r_t>());
-  STF_EQUAL(bs::std_(abs)(bs::Minf<T>()), bs::Inf<r_t>());
-  STF_IEEE_EQUAL(bs::std_(abs)(bs::Nan<T>()),  bs::Nan<r_t>());
```
Comment (jfalcou) [inline test/function/scalar/abs.cpp:66]: I'm a bit reluctant to change those. IEEE_EQUAL clearly means what it means when comparing to NaN.

**Review by jfalcou — CHANGES_REQUESTED** [review]
Discuss the IEEE_EQUAL changes then we'll merge

File: test/function/scalar/abs.cpp (hunk 1 above)
Comment (jtlap) [reply test/function/scalar/abs.cpp:66]: It means to redo everything  as the reversal is not so simple. 

All the way the cases where IEEE_EQUAL is full proof have exactly the same outcome than ULP_EQUAL with 0.5. 

Moreover to get back will leave IEEE_EQUAL where it's not sure that a change of proc/archi will not
need the other call.

Comment (jfalcou) [reply test/function/scalar/abs.cpp:66]: the arch argument is compelling enough. WIll merge

**Review by jfalcou — APPROVED** [review]

**Pull request merged**